gateway/graphql/market_data_resolvers.go

```diff
@@ -0,0 +1,529 @@
+}
+
+// END: MarketData resolver
+
+// BEGIN: Market Depth Resolver
+
+type myMarketDepthResolver VegaResolverRoot
+
+func (r *myMarketDepthResolver) LastTrade(ctx context.Context, md *types.MarketDepth) (*types.Trade, error) {
+	if md == nil {
+		return nil, errors.New("invalid market depth")
+	}
+
+	req := protoapi.LastTradeRequest{MarketId: md.MarketId}
+	res, err := r.tradingDataClient.LastTrade(ctx, &req)
+	if err != nil {
+		r.log.Error("tradingData client", logging.Error(err))
+		return nil, customErrorFromStatus(err)
+	}
+	return res.Trade, nil
+}
+
+func (r *myMarketDepthResolver) SequenceNumber(ctx context.Context, md *types.MarketDepth) (string, error) {
+	return strconv.FormatUint(md.SequenceNumber, 10), nil
+}
+
+func (r *myMarketDepthResolver) Market(ctx context.Context, md *types.MarketDepth) (*types.Market, error) {
+	return r.r.getMarketByID(ctx, md.MarketId)
+}
+
+type myMarketDepthResolverV2 myMarketDepthResolver
+
+func (r *myMarketDepthResolverV2) LastTrade(ctx context.Context, md *types.MarketDepth) (*MarketDepthTrade, error) {
+	if md == nil {
+		return nil, errors.New("invalid market depth")
+	}
+
+	req := protoapi.LastTradeRequest{MarketId: md.MarketId}
+	res, err := r.tradingDataClient.LastTrade(ctx, &req)
+	if err != nil {
+		r.log.Error("tradingData client", logging.Error(err))
+		return nil, customErrorFromStatus(err)
+	}
+	return &MarketDepthTrade{ID: res.Trade.Id, Price: res.Trade.Price, Size: strconv.FormatUint(res.Trade.Size, 10)}, nil
+}
+
+func (r *myMarketDepthResolverV2) SequenceNumber(ctx context.Context, md *types.MarketDepth) (string, error) {
+	return (*myMarketDepthResolver)(r).SequenceNumber(ctx, md)
+}
+
+// END: Market Depth Resolver
+
+// BEGIN: Market Depth Update Resolver
+
+type myMarketDepthUpdateResolver VegaResolverRoot
+
+func (r *myMarketDepthUpdateResolver) SequenceNumber(ctx context.Context, md *types.MarketDepthUpdate) (string, error) {
+	return strconv.FormatUint(md.SequenceNumber, 10), nil
+}
+
+func (r *myMarketDepthUpdateResolver) PreviousSequenceNumber(ctx context.Context, md *types.MarketDepthUpdate) (string, error) {
+	return strconv.FormatUint(md.PreviousSequenceNumber, 10), nil
+}
+
+func (r *myMarketDepthUpdateResolver) Market(ctx context.Context, md *types.MarketDepthUpdate) (*types.Market, error) {
+	return r.r.getMarketByID(ctx, md.MarketId)
+}
+
+type myMarketDepthUpdateResolverV2 myMarketDepthUpdateResolver
+
+func (r *myMarketDepthUpdateResolverV2) SequenceNumber(ctx context.Context, md *types.MarketDepthUpdate) (string, error) {
+	return (*myMarketDepthUpdateResolver)(r).SequenceNumber(ctx, md)
+}
+
+func (r *myMarketDepthUpdateResolverV2) PreviousSequenceNumber(ctx context.Context, md *types.MarketDepthUpdate) (string, error) {
+	return (*myMarketDepthUpdateResolver)(r).PreviousSequenceNumber(ctx, md)
+}
+
+// END: Market Depth Update Resolver
+
+func (r *mySubscriptionResolver) MarketsDepth(ctx context.Context, marketIds []string) (<-chan []*types.MarketDepth, error) {
```

Comment:
The last 2 function from here, seems to have pretty much the same body. any change to make it something generic?